routers/user/oauth_userinfo.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"fmt"
+	"net/http"
+
+	"code.gitea.io/gitea/modules/context"
+)
+
+// userInfoResponse represents a successful userinfo response
+type userInfoResponse struct {
+	Sub      string `json:"sub"`
+	Name     string `json:"name"`
+	Username string `json:"preferred_username"`
+	Email    string `json:"email"`
+	Picture  string `json:"picture"`
+}
+
+// InfoOAuth responds with OAuth formatted userinfo
+func InfoOAuth(ctx *context.Context) {
+	user := ctx.User
+	response := &userInfoResponse{
+		Sub:      fmt.Sprint(user.ID),
+		Name:     user.FullName,
```

Comment:
Sometimes fullname is empty, this could have a fallback if fullname is empty then use Name

Reply:
technically per spec, this isn't required, but if it is empty they recommend not including it. So I won't block on this.